src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -201,6 +211,13 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                         routeValues,
                         method: null,
                         htmlAttributes: null);
+
+                    //Append the fragment to action
+                    if (Fragment != null)
+                    {
+                        tagBuilder.Attributes["action"] += string.Concat("#" + Fragment);
+                        Console.WriteLine(tagBuilder.Attributes["action"]);
```

Comment:
Yup I wont :) I just wanted to get some ideas about what should be tested and formatting conventions. Maybe I shouldn't have created a PR...Sorry!